src/main/kotlin/org/rust/ide/annotator/fixes/FillFunctionArgumentsFix.kt

```diff
@@ -64,44 +63,10 @@ class FillFunctionArgumentsFix(element: PsiElement) : LocalQuickFixAndIntentionA
 
 private fun getParameterTypes(element: PsiElement): List<Ty?>? {
     return when (element) {
-        is RsCallExpr -> {
-            when (val target = (element.expr as? RsPathExpr)?.path?.reference?.resolve()) {
-                is RsFunction -> {
-                    val valueParameters = target.valueParameters.map { it.typeReference?.type }
-                    if (target.isMethod) {
-                        listOf(getSelfType(target, element)) + valueParameters
-                    }
-                    else {
-                        valueParameters
-                    }
-                }
-                is RsFieldsOwner -> target.tupleFields?.tupleFieldDeclList?.map { it.typeReference.type }
-                else -> null
-            }
-        }
+        is RsCallExpr -> (element.expr.type as? TyFunction)?.paramTypes
```

Comment:
I like such fixes very much when you mostly remove code and something becomes work better 👍

Reply:
Me too :) I think that I have tried using the `type` directly for the case of a temporary lambda and it didn't work, but I didn't realize that it's because the plugin doesn't support these lambda cases, but overall the function `type` inference works fine.